internal/consumer/repo.go

```diff
@@ -0,0 +1,59 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+type DbRepository interface {
+	SaveItem(item Item) error
+}
+
+type MongoRepository struct {
+	Client *mongo.Client
+	Ctx    context.Context
+}
+
+func (mr MongoRepository) SaveItem(item Item) error {
+	database := mr.Client.Database("hacker-news")
+	itemsCollection := database.Collection("items")
+
+	_, err := itemsCollection.InsertOne(mr.Ctx, item)
+	if err != nil {
+		return errors.Wrap(err, "Save item: ")
+	}
+	fmt.Printf("Inserted Record: %v \n", item.Id)
+
+	return nil
+}
+
+func ConnectDb(ctx context.Context) (*mongo.Client, error) {
+	url := fmt.Sprintf("mongodb://%s:%s@%s:%s", os.Getenv("DB_USER"), os.Getenv("DB_PASS"), os.Getenv("DB_NAME"), os.Getenv("DB_PORT"))
```

Comment:
I would check each of these variables exists and error if they don't rather than use them directly